src/platform/native.zig

```diff
@@ -571,7 +571,30 @@ pub const Platform = struct {
     }
 };
 
-pub const BackingTimer = std.time.Timer;
+pub const BackingTimer = struct {
+    start_time: f64 = undefined,
+    timer: std.time.Timer = undefined,
+
+    pub fn start() std.time.Timer.Error!BackingTimer {
+        return BackingTimer{ .time = try std.time.Timer.start(), .start_time = std.time.Instant.now() };
+    }
+
+    pub fn read(self: *BackingTimer) u64 {
+        return self.timer.read();
+    }
+
+    pub fn readTotal(self: *BackingTimer) u64 {
+        return std.time.Instant.now() - self.start_time;
```

Comment:
`std.time.Instant.now()` is not guaranteed to return monotonic time values. That is, it may return a time that is *before* `self.start_time` if a user e.g. sets their system clock back in time. In which case, this would overflow.

Instead of storing `start_time` and doing this, I suggest instead we just keep two `std.time.Timer` fields around (as they are guaranteed to be monotonic):

```diff
-start_time: f64 = undefined,
+start_timer: std.time.Timer = undefined,
 timer: std.time.Timer = undefined,
```

Then this `readTotal` function should just be `return self.start_time.read()`.